latex/fonts-tlwg.sty

```diff
@@ -19,68 +19,120 @@
-\DeclareOptionX{sfumpush}{\renewcommand{\thaisfdefault}{umpush}}
-\DeclareOptionX{sfloma}{\renewcommand{\thaisfdefault}{loma}}
-\DeclareOptionX{sfwaree}{\renewcommand{\thaisfdefault}{waree}}
-\DeclareOptionX{ttttype}{\renewcommand{\thaittdefault}{ttype}}
-\DeclareOptionX{ttttypist}{\renewcommand{\thaittdefault}{ttypist}}
-
-% Set default font
-\DeclareOptionX{kinnari}{\renewcommand{\thaifamilydefault}{kinnari}}
-\DeclareOptionX{garuda}{\renewcommand{\thaifamilydefault}{garuda}}
-\DeclareOptionX{norasi}{\renewcommand{\thaifamilydefault}{norasi}}
-\DeclareOptionX{laksaman}{\renewcommand{\thaifamilydefault}{laksaman}}
-\DeclareOptionX{loma}{\renewcommand{\thaifamilydefault}{loma}}
-\DeclareOptionX{purisa}{\renewcommand{\thaifamilydefault}{purisa}}
-\DeclareOptionX{sawasdee}{\renewcommand{\thaifamilydefault}{sawasdee}}
-\DeclareOptionX{ttype}{\renewcommand{\thaifamilydefault}{ttype}}
-\DeclareOptionX{ttypist}{\renewcommand{\thaifamilydefault}{ttypist}}
-\DeclareOptionX{umpush}{\renewcommand{\thaifamilydefault}{umpush}}
-\DeclareOptionX{waree}{\renewcommand{\thaifamilydefault}{waree}}
-
-\ProcessOptionsX\relax
-
-\DeclareRobustCommand{\thaitext}{%
-  \fontencoding{LTH}\fontfamily{\thaifamilydefault}\selectfont
-  \def\familydefault{\thaifamilydefault}%
-  \def\rmdefault{\thairmdefault}%
-  \def\sfdefault{\thaisfdefault}%
-  \def\ttdefault{\thaittdefault}%
-  \def\encodingdefault{LTH}}
-
-\DeclareRobustCommand{\latintext}{%
-  \fontencoding{\latinencoding}\fontfamily{\englishfamilydefault}\selectfont
-  \def\familydefault{\englishfamilydefault}%
-  \def\rmdefault{\englishrmdefault}%
-  \def\sfdefault{\englishsfdefault}%
-  \def\ttdefault{\englishttdefault}%
-  \def\encodingdefault{\latinencoding}}
+% Font options for XeTeX and LuaTeX mode
+\iffontstlwg@otf
+
+  % Font families with regular, oblique, bold, and bold-oblique.
+  \defaultfontfeatures[Garuda,Loma,Purisa,Sawasdee,TlwgMono,TlwgTypewriter,%
+                       TlwgTypist,TlwgTypo,Umpush,Waree]{
+      Scale           = \fontstlwg@scale ,
+      Extension       = .otf ,
+      ItalicFont      = *-Oblique ,
+      BoldFont        = *-Bold ,
+      BoldItalicFont  = *-BoldOblique ,
+      Script          = Thai
+  }
+  % Font families with regular, italic, bold, and bold-italic.
+  \defaultfontfeatures[Laksaman]{
+      Scale           = \fontstlwg@scale ,
+      Extension       = .otf ,
+      ItalicFont      = *-Italic ,
+      BoldFont        = *-Bold ,
+      BoldItalicFont  = *-BoldItalic ,
+      Script          = Thai
+  }
+  % Font families with regular, italic, oblique, bold, 
+  % bold-italic, and bold-oblique.
+  \defaultfontfeatures[Kinnari,Norasi]{
+      Scale           = \fontstlwg@scale ,
+      Extension       = .otf ,
+      ItalicFont      = *-Italic ,
+      SlantedFont     = *-Oblique ,
+      BoldFont        = *-Bold ,
+      BoldItalicFont  = *-BoldItalic ,
+      BoldSlantedFont = *-BoldOblique ,
+      Script          = Thai
+  }
```

Comment:
@srakrn
Surely, any XeLaTeX user can directly setup these TLWG fonts with `fontspec` by themselves. However, it is somehow tedious to manually and correctly defines roman/italics/slated variants of each of these fonts. Hence, this PR will simplify this setup process by pre-define these font features in advance,
and thus requires XeLaTeX users to do only two steps:
- `\usepackage{fonts-tlwg}`
- `\setsansfont{Loma}`, etc.